agent/src/main/scala/za/co/absa/atum/agent/model/Measure.scala

```diff
@@ -17,182 +17,156 @@
 package za.co.absa.atum.agent.model
 
 import org.apache.spark.sql.functions._
-import org.apache.spark.sql.types.{DecimalType, LongType, StringType}
+import org.apache.spark.sql.types.{DataType, DecimalType, LongType, StringType}
 import org.apache.spark.sql.{Column, DataFrame}
 import za.co.absa.atum.agent.core.MeasurementProcessor
 import za.co.absa.atum.agent.core.MeasurementProcessor.{MeasurementFunction, ResultOfMeasurement}
 import za.co.absa.atum.model.dto.MeasureResultDTO.ResultValueType
-import za.co.absa.spark.commons.implicits.StructTypeImplicits.StructTypeEnhancements
 
 /**
  *  Type of different measures to be applied to the columns.
  */
-sealed trait Measure extends MeasurementProcessor with MeasureType {
-  val controlCol: String
+sealed trait Measure {
+  val measureName: String
+  def controlColumns: Seq[String]
 }
 
-trait MeasureType {
-  val measureName: String
-  val resultValueType: ResultValueType.ResultValueType
+case class CustomMeasure(measureName: String, controlCols: Seq[String]) extends Measure {
+  override def controlColumns: Seq[String] = controlCols
 }
 
-object Measure {
+trait AtumMeasure extends Measure with MeasurementProcessor {
+  val resultValueType: ResultValueType.ResultValueType
+}
 
-  private val valueColumnName: String = "value"
+object AtumMeasure {
 
-  val supportedMeasures: Seq[MeasureType] = Seq(
-    RecordCount,
-    DistinctRecordCount,
-    SumOfValuesOfColumn,
-    AbsSumOfValuesOfColumn,
-    SumOfHashesOfColumn
+  val supportedMeasureNames: Seq[String] = Seq(
```

Comment:
why not to keep that? So we would offer strings as well as objects. No criticism, just want to know your reasoning

Reply:
I don't mind having the previous implementation once we really use that seq of objects. But until then it complicates matters (compilation) when one is refactoring the code.

Reply:
Okay